### PR TITLE
fix integrate sqrt(x**2)

### PR DESCRIPTION
**Pull request by eagleoflqj**
#### Brief description of what is fixed or changed

Risch is wrong for this case, leading `integrate` wrong.
```py
In [1]: integrate(sqrt(x**2), risch=True)
Out[1]: 
 2
x 
──
2 

In [2]: integrate(sqrt(x**2), heurisch=True)
Out[2]: 
     ____
    ╱  2 
x⋅╲╱  x  
─────────
    2    

In [3]: integrate(sqrt(x**2), manual=True)
Out[3]: 
     ____
    ╱  2 
x⋅╲╱  x  
─────────
    2    

In [4]: integrate(sqrt(x**2), meijerg=True)
Out[4]: 
⌠           
⎮    ____   
⎮   ╱  2    
⎮ ╲╱  x   dx
⌡           

In [5]: integrate(sqrt(x**2))
Out[5]: 
 2
x 
──
2 
```
#### Release Notes

* integrals
  * Fix integrate sqrt(x**2) in risch

### CODE REVIEW

File: sympy/integrals/risch.py [hunk 1]
```diff
@@ -311,7 +311,7 @@ def _rewrite_exps_pows(self, exps, pows, numpows,
         # do this at each pass (or else modify it to not do that).
 
         ratpows = [i for i in self.newf.atoms(Pow).union(self.newf.atoms(exp))
-            if (i.base.is_Pow or isinstance(i.base, exp) and i.exp.is_Rational)]
+                   if (isinstance(i.base, exp) and i.exp.is_Rational)]
```
[the remark this answers is not in the document]
Comment (eagleoflqj) [reply sympy/integrals/risch.py:314]:
```py
>>> type(Pow(E, x))
exp
```
I guess like many other places we don't need to handle unevaluated expression.

[the remark this answers is not in the document]
Comment (eagleoflqj) [reply sympy/integrals/risch.py:314]: With https://github.com/sympy/sympy/blob/f4691d30fa95522795890f3740e4cb39034c5b5c/sympy/functions/elementary/exponential.py#L211-L218
we already have
```py
>>> with _exp_is_pow(True):
...     isinstance(exp(x), exp)
... 
True
```

**Comment by oscarbenjamin** [comment]
@asmeurer does this look good to you?

**Comment by eagleoflqj** [comment]
I'm going to merge this as it's a Pareto improvement according to tests, and I can start to work on #23712.

**Comment by oscarbenjamin** [comment]
It looks good to me but can you add a release note.

**Pull request merged**